src/main/java/com/lob/net/ResponseGetter.java

```diff
@@ -275,11 +288,12 @@ private static String urlEncodePair(String key, String value) throws Unsupported
         String lobURL = String.format("%s%s", Lob.API_BASE_URL, url);
 
         if (type == APIResource.RequestType.MULTIPART) {
-            return makeMultipartConnectionRequest(method, clazz, lobURL, params, options);
+            return makeMultipartConnectionRequest(method, clazz, lobURL, data, options);
         }
 
-        String query = createQuery(params);
-        return makeURLConnectionRequest(method, clazz, lobURL, query, options);
+        String encodedData = createQuery(data);
```

Comment:
While I think this is fine for now, my general suggestion would be to move the construction of URL params to each individual resource class and just pass the URL down to this `_request` method. This gives each class the ability to specify what type of URL params are supported as well as how they should be added, encoded, translated, etc.

Also, I don't know this answer off the top of my head, but Is it safe to assume that all our endpoints only support one value for each parameter type?

Find endpoints come into question if you want to filter for multiple things in one request. For example, do we have anything that might be `https://api.lob.com/blah?filter=a&filter=b&filter=c`?

Reply:
You can pass multiple values for a certain key and the function will encode them properly in a manner that the Lob API will read them accordingly. While we do not support a query of `filter=a&filter=b` we do support the array notation of `filter[]=a&filter[]=b`. You can see a test for this [here](https://github.com/lob/lob-java/blob/49e153962afce10e08845bd2b55cb3746ab571e9/src/test/java/com/lob/net/ResponseGetterTest.java#L32-L53).

As for the params being encoded in their own resource class, is there a benefit of doing this? All resources are encoded in the same manner.

Reply:
Encoding might not be the best example, but creating chainable methods to add params in java is probably a better implementation.

What I've seen in the past is you have a request builder where you call all the chainable methods. Then at the end, you call a `build` method that will construct the request object to pass to a client to then make.

For example, instead of passing in a Map containing filterA and filterB, you'd have some pseudocode like the following:

```java
RequestBuilder<Verification> request = new VerificationRequestBuilder()
  .filterA(valueA)
  .filterB(valueB)
  .build();

Verification verification = client.sendRequest(request);
```

Since this is a time sensitive change, this isn't a blocking comment. Probably something we should address for the direction of our java library in the future.